superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -973,11 +974,14 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
     ]);
 
     if (!(info.file.originFileObj instanceof File)) return;
-    await importResource(
+    const dbId = await importResource(
       info.file.originFileObj,
       passwords,
       confirmedOverwrite,
     );
+    if (dbId) {
```

Comment:
I think reducing if clause makes sense to me
```
if (dbId && onDatabaseAdd) {
  onDatabaseAdd()
}
```

Reply:
Updated :) Thank you both.